### DISCUSSION TITLE
zfs-windows-2.2.6-rc8

### DISCUSSION THREAD

**Discussion by lundman** [openzfsonwindows/openzfs#408] [Development]
Feels good to fix that import BSOD finally.

Right, let me know what to work on next?

**Comment by guenther-alka** [comment]
rc8

-  Does this affects the BSOD problem with Aomei

wishlist
-  zfs send over netcat that terminates after replication
-  access to snaps ex in z:\tank\data\.zfs\snapshot\2 still not working (snap2 empty when creating a file, then snap2, then delete file)
an older snapshot 1: Windows 11 Explorer hangs when I try to open snap 1
-  ZFS snaps via Windows previous versions?

**Reply by lundman** [reply]
I could load Aomei, and even force ZFS ZVOL on, but alas, the reading of Registry happens too late in rc8 so that needs to be fixed.

> zfs send over netcat that terminates after replication

Yeah but since `cat | netcat` it isn't for us/ZFS to fix, it needs `netcat` to fix. Since `ncat` and `netcat` can talk to each other, just `alias netcat ncat` on Windows sending :) (okok, you need a shell wrapper to add --send-only)


> access to snaps ex

I'll check it again. If you have steps to do to make it happen might help.  It certainly was working


> ZFS snaps via Windows previous versions

Yeah be fun to look into. But first is stability.

Keep 'em coming

**Reply by lundman** [reply]
Ah mounting one snapshot was fine, the second snapshot mounted did not work well. So let's say it's a feature, one snapshot at a time!

**Reply by guenther-alka** [reply]
It seems when more than one snap is in the .zfs/snapshot folder, you can only open the last one in Explorer and this one is always empty.

**Comment by guenther-alka** [comment]
thanks!

yes, indeed, stability first
hope that snapfolder access from Explorer can work

btw
network replication terminates correctly with nc64 and the following settings

```
#####################
nc64 Linux -> Windows
#####################

Windows
C:\Users\me>\xampp\web-gui\ncat.win\nc64.exe -v -l -d -w 10 -p 53087 | zfs receive -v tank/pve
receiving full stream of tank/data@2 into tank/pve@2
received 2.42M stream in 3.06 seconds (810K/sec)
C:\Users\me>

Linux
root@pve:~# zfs snap tank/data@2
root@pve:~# zfs send tank/data@2 | pv | nc 192.168.2.65 53087
2.42MiB 0:00:01 [1.23MiB/s] [ <=>
root@pve

###############################################
On Windows, tank/pve and tank/pve@2 is created
replication terminates  only with -l -d -w10 on Windows
##############################################


#####################
nc64 Windows -> Linux
#####################
C:\Users\me>zfs send tank/data2@1 | \xampp\web-gui\ncat.win\nc64.exe -w 10 192.168.2.71 53087
C:\Users\me>

Linux
root@pve~#nc -l -p 53087 | zfs receive tank/win
root@pve~#

################################################
On Linux, tank/win and tank/win@1 is created
replication terminates only with -w10 on Windows
################################################
```
Do not forget to use cmd as pipes have problems in Powershell
In powershell start pipe commands like:

Start-Process cmd -ArgumentList "/c", "\xampp\web-gui\data\cs_server\tools\nc\win\nc64 -l -w 20 -d -p 51437 | zfs receive -Fv tank/prox  >> \xampp\web-gui\data\cs_server\tmp\remote.log" -WindowStyle minimized

**Reply by guenther-alka** [reply]
the snapshot problem seems more serious

When I import the pool:
Explorer mounts a driveletter ex L: that shows current filesystems but also older ones that were deleted

When I try to open a snap in .zfs/snapshot (beside last one) Explorer hangs
Last one is empty but should not..

When I try to export the pool to solve the problem, ZFS seems to hang, then I get a BSOD (openzfs.sys)
https://drive.google.com/file/d/19i50MxcDpm3NeFNBL7_KPU2_CyDByRJQ/view?usp=sharing

**Reply by lundman** [reply]
OK had a busy loop when trying to list more than 1 snapshot.
![Screenshot 2024-10-30 at 12 14 29 (2)](https://github.com/user-attachments/assets/beb6e70a-b0f5-4ed9-a2c4-a0b6b8711097)

**Reply by lundman** [reply]
2c7fc76

**Comment by lundman** [comment]
OK whats next.. any more BSODs?
